config/config.go

```diff
@@ -0,0 +1,61 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+type TLSConfig struct {
+	// Path to x509 certificate
+	CertPath string `json:"certPath"`
+	// Path to private key
+	KeyPath string `json:"keyPath"`
+	// Path to server CA certificate
+	CACertPath string `json:"caPath"`
+	// Disable tls host name verification (tls must be enabled)
+	DisableHostVerification bool `json:"disableHostVerification"`
+	// Override for target server name
+	ServerName string `json:"serverName"`
+}
+
+type ClusterConfig struct {
+	// host:port for Temporal frontend service
+	Address string `json:"address"`
+	// Web UI Link
+	WebAddress string `json:"webAddress"`
+	// Temporal workflow namespace (default: "default")
+	Namespace string `json:"namespace"`
+	// Headers provider plugin executable name
+	HeadersProvider string `json:"headersProvider"`
+	// Data converter plugin executable name
+	DataConverter string     `json:"dataConverter"`
+	TLS           *TLSConfig `json:"tls,omitempty"`
+	// Any additional environment variables that are needed
+	Environment map[string]string `json:"additional,omitempty"`
+}
+
+type Config struct {
+	ActiveContext string `json:"active"`
+	// Map of context names to cluster configuration
+	Contexts map[string]*ClusterConfig `json:"contexts"`
+}
+
+func (c ClusterConfig) GetTLS() TLSConfig {
+	if c.TLS == nil {
+		return TLSConfig{}
+	}
+	return *c.TLS
+}
+
+func GetDefaultConfigPath() (string, error) {
```

Comment:
Nit: please include godoc for public functions/methods.

```suggestion
// GetDefaultConfigPath returns the path to the current user's default tctx config file.
// On unix systems, this will be `$XDG_CONFIG_HOME/tctx/config.json`.
func GetDefaultConfigPath() (string, error) {
```